gcp-cloud-run/locals.tf

```diff
@@ -0,0 +1,51 @@
+locals {
+  pvault_region           = coalesce(var.pvault_region, var.default_region)
+  pvault_db_socket_folder = "/cloudsql"
+
+  # Admin API key
+  pvault_admin_api_key_generate         = var.pvault_admin_api_key.secret_id == null
+  pvault_admin_api_key_rotation_enabled = var.pvault_admin_api_key.rotate_days > 0 && local.pvault_admin_api_key_generate
+  pvault_admin_api_key_secret           = local.pvault_admin_api_key_generate ? google_secret_manager_secret.admin_api_key[0] : data.google_secret_manager_secret.admin_api_key[0]
+
+  # Generate env variables
+  pvault_env = merge(
+    {
+      PVAULT_LOG_CUSTOMER_IDENTIFIER = var.pvault_log_customer_identifier
+      PVAULT_LOG_CUSTOMER_ENV        = var.pvault_log_customer_env
+      PVAULT_LOG_CUSTOMER_REGION     = var.pvault_region
+      PVAULT_TLS_ENABLE              = "false"
+      PVAULT_DB_REQUIRE_TLS          = "false"
+      PVAULT_DB_HOSTNAME             = "${local.pvault_db_socket_folder}/${module.postgresql-db.instance_connection_name}"
+      PVAULT_DB_NAME                 = var.cloudsql_name
+      PVAULT_DB_USER                 = var.cloudsql_username
+      PVAULT_SERVICE_LICENSE         = var.pvault_service_license
+      PVAULT_KMS_URI                 = "gcp-kms://${google_kms_crypto_key.vault-encryption-key.key_ring}/cryptoKeys/${google_kms_crypto_key.vault-encryption-key.name}"
+      PVAULT_DEVMODE                 = tostring(var.pvault_devmode)
+      # Update API key on restart
+      PVAULT_SERVICE_OVERRIDE_ADMIN_API_KEY_ON_RESTART = local.pvault_admin_api_key_rotation_enabled || !local.pvault_admin_api_key_generate
+    },
+    var.pvault_env_vars,
```

Comment:
Is it cool that the user can override the dedicated variables with own env vars? pvault_env_vars to take precedence

Reply:
I've been thinking bout about it, in the previous PR I kept it at lower priority, but there are cases where the user might want override something (e.g. `PVAULT_SERVICE_OVERRIDE_ADMIN_API_KEY_ON_RESTART`, `PVAULT_DB_REQUIRE_TLS`).

Lets see how it goes, if there is undesired outcomes, I propose to split it in two parts (Enforced and Default vars).